Makefile

```diff
@@ -0,0 +1,99 @@
+# List all independent nodes, separated with escaped newlines
+NODES = \
+serial_bridge 
+
+# Code running onboard
+ONBOARD = onboard_tmp
+
+LCM_TYPES_DIR = biped_lcm
+LCM_PACKAGE_NAME = biped_lcm
+LCM_TARGET_JAR = biped_lcm.jar
+
+MATLAB_JAVA = 1.7
+
+# TODO: make this independent of DRAKE_DISTRO
+LCM_JAR = $(DRAKE_DISTRO)/build/install/share/java/lcm.jar
+
+#----------------------------------------------
+
+export CXX=g++
+export CXXFLAGS=-std=c++11 -Wall
+
+BUILD_DIR = build
+NODE_TARGET_NAME = main
+
+#path to this makefile, i.e. to the top level biped dir
+MKFILE_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+LCM_DEFS = $(wildcard $(LCM_TYPES_DIR)/*.lcm)
+
+default: all
+
+# -----------Build LCM CPP:
+LCM_B_DIR = $(BUILD_DIR)/$(LCM_TYPES_DIR)
+export LCM_BDIR_ABS=$(MKFILE_DIR)$(LCM_B_DIR)
+
+LCM_HEADERS = $(patsubst $(LCM_TYPES_DIR)/%.lcm, $(LCM_B_DIR)/%.hpp, $(LCM_DEFS))
+$(LCM_B_DIR)/%.hpp: $(LCM_TYPES_DIR)/%.lcm
+	lcm-gen --cpp --cpp-std=c++11 --cpp-hpath=$(LCM_B_DIR) $^
+
+#----------- Build LCM Java: 
+LCM_JAVA_DIR = $(LCM_B_DIR)/$(LCM_PACKAGE_NAME)
+# Build .java and .class files
+$(LCM_JAVA_DIR)/%.class: $(LCM_TYPES_DIR)/%.lcm
```

Comment:
Good call, you're right. Turns out lcm-gen makes the target directories anyway. Got rid of it and pushed it to master.